internal/release/render/render.go

```diff
@@ -200,6 +205,44 @@ func HydrateValues(release *v1alpha1.Release, mappings *config.ValueMapping) (ma
 		return nil, err
 	}
 
+	result, err = unifyValues(result, chart)
+	if err != nil {
+		return nil, fmt.Errorf("unifying with chart schema: %w", err)
+	}
+
+	return result, nil
+}
+
+func unifyValues(values map[string]any, chart *helm.ChartFS) (map[string]any, error) {
+	if chart == nil {
+		return values, nil
+	}
+
+	rawSchema, err := chart.ReadFile("values.cue")
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return values, nil
+		}
+		return nil, fmt.Errorf("reading values.cue: %w", err)
+	}
+
+	runtime := cuecontext.New()
+
+	schema := runtime.
+		CompileBytes(rawSchema).
+		LookupPath(cue.MakePath(cue.Def("#values")))
+
+	unified := schema.Unify(runtime.Encode(values))
+
+	if err := unified.Validate(cue.Final(), cue.Concrete(true)); err != nil {
+		return nil, xerr.MultiErrFrom("validating values", AsErrorList(cueerrors.Errors(err))...)
```

Comment:
Shouldn't that be "unifying" instead of "validating"?
```suggestion
		return nil, xerr.MultiErrFrom("unifying values", AsErrorList(cueerrors.Errors(err))...)
```

Reply:
This is a validation error!

Reply:
Effectivement, désolé, j'avais mal regardé! 😅